modules/dag-l1/src/main/scala/org/tessellation/dag/l1/domain/block/BlockService.scala

```diff
@@ -0,0 +1,17 @@
+package org.tessellation.dag.l1.domain.block
+
+import org.tessellation.dag.domain.block.{BlockReference, DAGBlock}
+import org.tessellation.security.Hashed
+import org.tessellation.security.hash.ProofsHash
+import org.tessellation.security.signature.Signed
+
+import eu.timepit.refined.types.numeric.PosInt
+
+trait BlockService[F[_]] {
+  def areParentsAccepted(block: DAGBlock): F[Map[BlockReference, Boolean]]
+  def accept(hashedBlock: Hashed[DAGBlock]): F[Unit]
+  def store(hashedBlock: Hashed[DAGBlock]): F[Unit]
+  def getWaiting: F[Map[ProofsHash, Signed[DAGBlock]]]
+  def handleTipsUpdate(hashedBlock: Hashed[DAGBlock]): F[Unit]
```

Comment:
btw, shouldn't `handleTipsUpdate` be private and executed internally by `accept`?

Reply:
Was thinking about it. I intended it to run synchronously as a next step in block acceptance flow (in the upcoming PRs), so semantically it will be the same (it could run in acceptance flow or here as a next step after accept). For now I'm not sure if we should hide it behind the interface, as I'm not sure how will the whole redownload flow work. The most appealing scenario is that we would only accept blocks one after the other and the tips will be calculated based on these dynamically no matter the case (normal acceptance, acceptance after redownload). But it could also be the case that during redownload we will only want a certain pool of blocks to go through the tip updating. I wouldn't hide it for now.

Reply:
so I'd create a second function for redownload specifically that will override some checks and force tips update, wdyt?

Reply:
I guess that depends on whether redownload will be able to reuse this one or not. If yes I would keep it in the interface to not have any functionality duplication, if no and redownload will need a separate function then we can hide it. Don't know at this point. I can create a ticket to reevaluate at some point.

Reply:
Ok, please do so